### ISSUE TITLE
[GDNative] compilation error between godot_gdnative_api_struct and gdnative_api_struct.gen.h

### ISSUE THREAD

**Issue by touilleMan** [godotengine/godot#13142]
GDNative headers seems to be broken, compiling godot-python creates the following errors:

```
➜  godot-python git:(master) ✗ scons backend=pypy backend_path=pypy3.5-5.9-beta-linux_x86_64-portable/ 
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
gcc -o pythonscript/pythonscript.os -c -DBACKEND_PYPY -I pypy3.5-5.9-beta-linux_x86_64-portable/include -Ipythonscript -std=c11 -pthread -DDEBUG=1 -fwrapv -Wall -g -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Bsymbolic-functions -Wformat -Werror=format-security -fdiagnostics-color=always -fPIC -Igodot/modules/gdnative/include pythonscript/pythonscript.c
In file included from pythonscript/pythonscript.h:4:0,
                 from pythonscript/pythonscript.c:6:
pythonscript/pythonscript.c: In function 'godot_gdnative_init':
godot/modules/gdnative/include/gdnative_api_struct.gen.h:15:30: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
 _gdnative_wrapper_api_struct = options->api_struct; \
                              ^
pythonscript/pythonscript.c:61:2: note: in expansion of macro 'GDNATIVE_API_INIT'
  GDNATIVE_API_INIT(options);
  ^
godot/modules/gdnative/include/gdnative_api_struct.gen.h:16:49: error: 'godot_gdnative_api_struct {aka const struct godot_gdnative_api_struct}' has no member named 'num_extensions'
 for (int i = 0; i < _gdnative_wrapper_api_struct->num_extensions; i++) { \
                                                 ^
pythonscript/pythonscript.c:61:2: note: in expansion of macro 'GDNATIVE_API_INIT'
  GDNATIVE_API_INIT(options);
  ^
godot/modules/gdnative/include/gdnative_api_struct.gen.h:17:38: error: 'godot_gdnative_api_struct {aka const struct godot_gdnative_api_struct}' has no member named 'extensions'
  switch (_gdnative_wrapper_api_struct->extensions[i]->type) { \
                                      ^
pythonscript/pythonscript.c:61:2: note: in expansion of macro 'GDNATIVE_API_INIT'
  GDNATIVE_API_INIT(options);
  ^
godot/modules/gdnative/include/gdnative_api_struct.gen.h:19:123: error: 'godot_gdnative_api_struct {aka const struct godot_gdnative_api_struct}' has no member named 'extensions'
    _gdnative_wrapper_nativescript_api_struct = (godot_gdnative_ext_nativescript_api_struct *) _gdnative_wrapper_api_struct->extensions[i]; \
                                                                                                                           ^
pythonscript/pythonscript.c:61:2: note: in expansion of macro 'GDNATIVE_API_INIT'
  GDNATIVE_API_INIT(options);
  ^
In file included from pythonscript/pythonscript.h:4:0,
                 from pythonscript/pythonscript.c:6:
godot/modules/gdnative/include/gdnative_api_struct.gen.h:22:123: error: 'godot_gdnative_api_struct {aka const struct godot_gdnative_api_struct}' has no member named 'extensions'
    _gdnative_wrapper_pluginscript_api_struct = (godot_gdnative_ext_pluginscript_api_struct *) _gdnative_wrapper_api_struct->extensions[i]; \
                                                                                                                           ^
pythonscript/pythonscript.c:61:2: note: in expansion of macro 'GDNATIVE_API_INIT'
  GDNATIVE_API_INIT(options);
  ^
godot/modules/gdnative/include/gdnative_api_struct.gen.h:25:107: error: 'godot_gdnative_api_struct {aka const struct godot_gdnative_api_struct}' has no member named 'extensions'
    _gdnative_wrapper_arvr_api_struct = (godot_gdnative_ext_arvr_api_struct *) _gdnative_wrapper_api_struct->extensions[i]; \
                                                                                                           ^
pythonscript/pythonscript.c:61:2: note: in expansion of macro 'GDNATIVE_API_INIT'
  GDNATIVE_API_INIT(options);
  ^
pythonscript/pythonscript.c:137:1: error: expected 'while' before 'void'
 void godot_gdnative_singleton() {
 ^
pythonscript/pythonscript.c:142:1: error: expected declaration or statement at end of input
 }
 ^
scons: *** [pythonscript/pythonscript.os] Error 1
scons: building terminated because of errors.
```

Trouble is `godot_gdnative_api_struct` doesn't correspond to what GDnative's `SCsub` expect when generating `gdnative_api_struct.gen.h`

https://github.com/godotengine/godot/blob/20b07a1fb55c31d326c3f9c36d1276eb1bcf15e6/modules/gdnative/include/gdnative/gdnative.h#L239-L243

https://github.com/godotengine/godot/blob/20b07a1fb55c31d326c3f9c36d1276eb1bcf15e6/modules/gdnative/SCsub#L27-L32

@karroffel I guess you forgot to add you modifications to `SCsub` when commiting 0865365e2 ;-)

**Comment by karroffel** [comment]
Oh yeah, oops. Will fix tonight or tomorrow! Otherwise feel free to make a patch 😄

**Comment by akien-mga** [comment]
Fixed by #13149.
